fix(navigation-block): reduce subnavigation arrow padding

devfle opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#pull-requests. -->

## Description
<!-- Please describe what you have changed or added -->
I currently work very intensively with full site editing. In the navigation block I noticed that the padding between the subnavigation point and the arrow icon is much too large. I have now adjusted this in the editor and on the published page.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
I have built the Gutenberg plugin and installed it into my local wordpress installation. Afterwards I checked my changes in different browsers and could not find any errors. I tested it with many, few and long navigation points. I also checked on mobile viewports.

1. Build Gutenberg Plugin
2. Inserted Navigation Block
3. Added many, few and long nav points
4. testet on diffrent browsers
5. resized browser / navigation few times

## Screenshots <!-- if applicable -->
**Before:**
![before-site](https://user-images.githubusercontent.com/52854338/88426180-043adb80-cdf1-11ea-83e7-d79344465354.png)

![before-editor](https://user-images.githubusercontent.com/52854338/88426505-993dd480-cdf1-11ea-80da-f4db726cd39f.png)


**After:**
![after-website](https://user-images.githubusercontent.com/52854338/88426254-2a607b80-cdf1-11ea-9c20-5d63d525b881.png)

![after-editor](https://user-images.githubusercontent.com/52854338/88426479-8b884f00-cdf1-11ea-8902-715443cc5086.png)






## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

Commit:
fix(navigation-block): reduce spacing between subnav arrow and nav point

devfle commented:
Edit: I tested both navigation blocks -> horizontal and vertical.

devfle commented:
Hi @talldan, @Soean, @ajitbohra - Just wanted to ask if someone can have a look. :-) Thank you!

tellthemachines commented on packages/block-library/src/navigation-link/style.scss:
```diff
@@ -29,6 +29,9 @@
 // Styles for submenu flyout
 .has-child {
 	$navigation-vertical-padding: $grid-unit-10 * 0.75;
+	> a {
```
If you target `.wp-block-navigation-link__content` instead of `a` in this file, it will apply to both editor and front end, so you won't need the change in `editor.scss` 🙂

devfle replied:
Thank you very much! I will work on it 😄 !

tellthemachines reviewed (commented):
The units used for padding have changed since this PR was opened: we're now using `em` instead of `px`. That's what is causing the merge conflicts btw 😅 . Best approach is to rebase master and then convert your changes to `em`.

Commit:
style(navigation-block): reduce padding on subnav arrow

devfle commented:
@tellthemachines Thanks a lot for your help. I rebased the project, converted the px values to em values and removed the redundant code.

I have tested the whole thing again. Looks good so far.

tellthemachines reviewed (approved):
Thanks for the quick turnaround! Looks good 🚀

kevin940726 commented on packages/block-library/src/navigation-link/style.scss:
```diff
@@ -28,6 +28,10 @@
 
 // Styles for submenu flyout
 .has-child {
+	$navigation-vertical-padding: $grid-unit-10 * 0.75;
```
Hmm, why do we need this 🤔 ? Is it used anywhere?

devfle replied:
The variable is actually no longer needed. It was also removed in the previous PR.

I have now removed it too and tested everything again. So far everything looks good. Thanks!

Commit:
style(navigation-block): remove unused scss var
